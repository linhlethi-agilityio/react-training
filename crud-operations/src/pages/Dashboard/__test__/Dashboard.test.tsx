import { render } from '@testing-library/react';

// Components
import DashboardPage from '..';

// Mocking dependencies
jest.mock('@hooks', () => ({
  usePayments: () => jest.fn(),
  useStudents: () => jest.fn(),
}));

jest.mock('jose', () => ({}));

jest.mock('@components', () => ({
  CardItem: () => <div />,
}));

jest.mock('@constants', () => ({
  ENVS: {
    VITE_API_ENDPOINT: 'import.meta.env.VITE_API_ENDPOINT',
    VITE_SECRET_KEY: 'import.meta.env.VITE_SECRET_KEY',
  },
}));

describe('DashboardPage Component', () => {
  test('renders correctly', () => {
    const { container } = render(<DashboardPage />);

    expect(container).toMatchSnapshot();
  });

  // Test other aspects similarly...
});
