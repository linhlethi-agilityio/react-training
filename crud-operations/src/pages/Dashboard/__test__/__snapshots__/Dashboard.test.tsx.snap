// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DashboardPage Component renders correctly 1`] = `
<div>
  <div
    class="css-16azfwd"
  >
    <div />
    <div />
    <div />
    <div />
  </div>
</div>
`;
